src/invocation/ClusterService.ts

```diff
@@ -58,29 +58,43 @@ class MemberListSnapshot {
 const EMPTY_SNAPSHOT = new MemberListSnapshot(-1, new Map<string, MemberImpl>(), []);
 const INITIAL_MEMBERS_TIMEOUT_IN_MILLIS = 120 * 1000; // 120 seconds
 
+
+interface ClientWithCM {
+    getConnectionManager(): ClientConnectionManager;
```

Comment:
This `ClusterService` <-> `ClientConnectionManager` dependency is the most tricky one. Luckily, both are internal modules (except for the `Cluster` interface part of `ClusterService`), so we can move things around to break the cycle.

I'll list some draft ideas below. Obviously, you may come up with a better solution for this problem.

Say, `ClusterService#getLocalClient` method which depends on `connectionManager` can be moved into `HazelcastClient`.

`ClusterService#detectMembershipEvents` could accept `ClientConnectionManager` instance when it's invoked from `ClusterViewListenerService`.

Reply:
These are great ideas! I applied them to the code